rflx/model.py

```diff
@@ -560,7 +560,18 @@ def __type_constraints(self, expr: Expr) -> Sequence[Expr]:
             if isinstance(t, Scalar)
         ]
 
-        return [c for n, t in scalar_types for c in t.constraints(name=n, proof=True)]
+        aggregate_size: List[Expr] = [
+            Equal(Length(r.left), Length(r.right))
+            for r in expr.findall(
+                lambda x: isinstance(x, Equal)
+                and (isinstance(x.left, Aggregate) or isinstance(x.right, Aggregate))
+            )
+            if isinstance(r, Equal)
```

Comment:
`NotEqual` should be also considered.

Reply:
What would be a sensible way of handling `NotEqual` - raise an error if it's statically false?

Reply:
Yes, I think so.

Reply:
Done in 2ef827d